provider/github/watcher_test.go

```diff
@@ -133,16 +145,18 @@ func (s *WatcherTestSuite) TestWatch_WithError() {
 
 	s.NoError(err)
 
+	var wg sync.WaitGroup
+
 	err = w.Watch(context.TODO(), func(e lookout.Event) error {
-		switch e.Type() {
-		case pb.ReviewEventType:
-			s.Equal("96de6d9d321aa94faa0f053c72e2684a44961e5b", e.ID().String())
-		case pb.PushEventType:
-			s.Equal("d1f57cc4e520766576c5f1d9e7655aeea5fbccfa", e.ID().String())
-		}
+		wg.Add(1)
+
+		s.Equal("d1f57cc4e520766576c5f1d9e7655aeea5fbccfa", e.ID().String())
+
 		return fmt.Errorf("foo")
 	})
 
+	wg.Wait()
```

Comment:
I don't see `wg.Done()` anywhere. This line waits forever.

Reply:
The funny thing is: `Done` was there, I did a small refactor and it got lost... But it does not wait forever because `Watch` finishes when there is an error unmarshalling the `PushEvent` (changed a string to an int).

Then `s.Error(err, "foo")` is not actually checking the error message for "foo", it should be `s.EqualError(err, "foo")`. So, the test was passing...